src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+                                        CSVFormat.DEFAULT)) {
+                                		pw.printRecord(getColumnNames());
+                                		int rowCount = getTable().getRowCount(); 
+                                		for (int row = 0; row < rowCount; row++) {
+                                        		pw.printRecord(getRowCells(row));
+                                		}		
+                        	} catch (Exception ex) {
+                                	success = false;
+                                	JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+                                                Constant.messages.getString("export.button.error") + "\n"
+                                                                + ex.getMessage());
+                                	LOGGER.error("Export Failed: " + ex.getMessage(), ex);
+                        	}
+			if (success) {
+                                JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+                                                Constant.messages.getString("export.button.success"));
+                        }
+                	}
+
+			}
+		});
+		pack();
+		return result;
+	 }
+	
+	protected List<String> getColumnNames() {
+                List<String> columnNamesList = new ArrayList<>();
+                for (int col = 0; col < getTable().getColumnCount(); col++) {
+                        columnNamesList.add(getTable().getColumnModel().getColumn(col).getHeaderValue().toString());
+                }
+                return columnNamesList;
+        }
+
+	protected List<Object> getRowCells(int row) {
+                List<Object> cells = new ArrayList<>();
+                for (int col = 0; col < getTable().getColumnCount(); col++) {
+                        Object value = getTable().getValueAt(row, col);
+                        cells.add(value == null ? "" : value.toString());
+                }
+                return cells;
+        }
+
+	protected JTable getTable() {
+                return exportTable;
+        }
+
+        public void setTable(JTable table) {
+                this.exportTable = table;
+        }
+	
+	 private JPanel getJPanel() {
+		//this.setLayout(new BorderLayout());
+		Panel.setSize(new Dimension(600, 600));
+		Panel.setVisible(true);
+		tp.add("Field Enumeration", jPanel);
+		jPanel.setLayout(new GridBagLayout());
+		c1.fill = GridBagConstraints.HORIZONTAL;
+		jPanel.setSize(new Dimension(600, 600));
+		c1.gridx = 0;
+                c1.gridy = 0;
+		JLabel l1 = new JLabel("URL Form:");
+		l1.setFont(font);
+		jPanel.add(l1, c1);
+		c1.anchor = GridBagConstraints.NORTH;
+		c1.gridx = 1;
+                c1.gridy = 0;
+		jPanel.add(printURL, c1);
+		c1.gridwidth = 2;
+		c1.gridx = 0;
+                c1.gridy = 1;
+		JLabel l2 = new JLabel("Enter Regular Expression or status code");
```

Comment:
My earlier comment: https://github.com/zaproxy/zap-extensions/pull/951/files#r136881128

Should apply from here I guess. Re; i18n strings and meaningful variable names.

Reply:
Apologies for repeated mistakes! I have used i18n strings and changed variables :)